fixtures/browser.py

```diff
@@ -30,8 +30,25 @@ def pytest_runtest_setup(item):
 
 def pytest_exception_interact(node, call, report):
     from fixtures.artifactor_plugin import SLAVEID
+    from httplib import BadStatusLine
+    from socket import error
+    from utils.browser import WharfFactory
+
+    import urllib2
+
     name, location = get_test_idents(node)
     val = safe_string(call.excinfo.value.message).decode('utf-8', 'ignore')
+
+    if isinstance(call.excinfo.value, urllib2.URLError) or \
+       isinstance(call.excinfo.value, BadStatusLine) or \
+       isinstance(call.excinfo.value, error):
+        from utils.browser import manager
+        if isinstance(manager.factory, WharfFactory):
```

Comment:
browser restarts due to a rest call failing thats not inside selenium, but any other api using urllib or sockets and not having propper exception layering